lib/proxy/validations.rb

```diff
@@ -1,8 +1,14 @@
+require 'ipaddr'
+require 'resolv'
+
 module Proxy::Validations
   MAC_REGEXP_48BIT = /\A([a-f0-9]{1,2}:){5}[a-f0-9]{1,2}\z/i
   MAC_REGEXP_64BIT = /\A([a-f0-9]{1,2}:){19}[a-f0-9]{1,2}\z/i
 
-  class Error < RuntimeError; end
```

Comment:
Please keep this one since it's the public API and used (as `Proxy::Validations::Error`). Also let the other exceptions inherit from this so you can use `rescue Proxy::Validations::Error` to catch any validation error.